api/task_delete.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/consul-terraform-sync/logging"
+)
+
+// deleteTask deletes an existing task and its events. Does not delete
+// if the task is active.
+func (h *taskHandler) deleteTask(w http.ResponseWriter, r *http.Request) {
+	taskName, err := getTaskName(r.URL.Path, taskPath, h.version)
+	logger := logging.FromContext(r.Context()).Named(deleteTaskSubsystemName)
+	if err != nil {
+		logger.Trace("bad request", "error", err)
+		jsonErrorResponse(r.Context(), w, http.StatusBadRequest, err)
+		return
+	}
+
+	if taskName == "" {
```

Comment:
Since the task name is part of the path, if we routed to this handler without a taskName that would actually be an error in how the programmer setup routing. Since it a programmer error, and not one a user could perform, we might want to panic instead? Or else not include this check?

Reply:
Yeah, I think I'll remove the check. When I wrote the code, I had more in mind the original handler since the openapi stuff was still WIP, so that's why I needed this check! But now that it's specifically for the open api handler, it doesn't seem necessary.